src/backend/storage/file/buffile.c

```diff
@@ -1012,6 +1034,10 @@ BufFileStartCompression(BufFile *file)
 {
 	ResourceOwner oldowner;
 	size_t ret;
+	ZSTD_customMem customMem;
+
+	customMem.customAlloc = customAlloc;
+	customMem.customFree = customFree;
```

Comment:
Is there a missing line here?
```c
customMem.opaque = file->resowner;
```

Reply:
Actually, we don't need this for deallocation function, as we just pass a pointer to be freed

Reply:
What if zstd on deallocation calls an allocation callback?

Reply:
No, out custom allocator is used in `ZSTD_customFree` which just passes `opaque` to our custom deallocator. In a case when we have no custom `free`, zstd calls free on `ptr` and that's all. The reason we have `opaque` as a parameter is backwards compatibility and it's used by zstd itself when legacy api is used